Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -1103,160 +1034,182 @@ public virtual bool Obey() =>
+    
             case OrderType.Guard:
-                {
-                    m_Mobile.ControlMaster.RevealingAction();
-                    m_Mobile.SetCurrentSpeedToActive();
-                    m_Mobile.PlaySound(m_Mobile.GetIdleSound());
-                    m_Mobile.Warmode = true;
-                    m_Mobile.Combatant = null;
-                    m_Mobile.ControlMaster.SendLocalizedMessage(1049671, m_Mobile.Name); // ~1_PETNAME~ is now guarding you.
-                    break;
-                }
-
+                HandleGuardOrder();
+                break;
+    
             case OrderType.Attack:
-                {
-                    m_Mobile.ControlMaster.RevealingAction();
-                    m_Mobile.SetCurrentSpeedToActive();
-                    m_Mobile.PlaySound(m_Mobile.GetIdleSound());
+                HandleAttackOrder();
+                break;
+    
+            case OrderType.Follow:
+                HandleFollowOrder();
+                break;
 
-                    m_Mobile.Warmode = true;
-                    m_Mobile.Combatant = null;
-                    break;
-                }
+            case OrderType.Rename:
+                HandleRenameOrder();
+                break;
+        }
+    }
+    
+    private void HandlePassiveOrder()
+    {
+        if (m_Mobile.Warmode)
+        {
+            m_Mobile.Warmode = false;
+        }
+        if (m_Mobile.Combatant != null)
+        {
+            m_Mobile.Combatant = null;
+        }
+        if (m_Mobile.FocusMob != null)
+        {
+            m_Mobile.FocusMob = null;
+        }
+        if (m_Mobile.CurrentSpeed != m_Mobile.PassiveSpeed)
+        {
+            m_Mobile.CurrentSpeed = m_Mobile.PassiveSpeed;
+        }
 
-            case OrderType.Patrol:
-                {
-                    m_Mobile.ControlMaster.RevealingAction();
-                    m_Mobile.SetCurrentSpeedToActive();
-                    m_Mobile.PlaySound(m_Mobile.GetIdleSound());
-                    m_Mobile.Warmode = false;
-                    m_Mobile.Combatant = null;
-                    break;
-                }
+        m_Mobile.PlaySound(m_Mobile.GetIdleSound());
+    }
+    
+    private void HandleGuardOrder()
+    {
+        if (m_Mobile.ControlMaster == null || !m_Mobile.ControlMaster.Alive)
+        {
+            return;
+        }
 
-            case OrderType.Release:
-                {
-                    m_Mobile.ControlMaster.RevealingAction();
-                    m_Mobile.SetCurrentSpeedToPassive();
-                    m_Mobile.PlaySound(m_Mobile.GetIdleSound());
-                    m_Mobile.Warmode = false;
-                    m_Mobile.Combatant = null;
-                    break;
-                }
+        if (!m_Mobile.Warmode)
+        {
+            m_Mobile.Warmode = true;
+        }
+        if (m_Mobile.FocusMob != null)
+        {
+            m_Mobile.FocusMob = null;
+        }
+        if (m_Mobile.CurrentSpeed != m_Mobile.PassiveSpeed)
```

Comment:
All of these checks are handled for us in the setter. We can get rid of them for all of these properties.